tests/components/switcher_kis/test_init.py

```diff
@@ -0,0 +1,61 @@
+"""Test cases for the switcher_kis component."""
+
+from typing import Any, Generator
+
+from homeassistant.components.switcher_kis import (DOMAIN, DISCOVERY_DEVICE)
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from tests.common import assert_setup_component
+
+from .consts import (
+    DUMMY_AUTO_OFF_SET, DUMMY_DEVICE_ID, DUMMY_DEVICE_NAME,
+    DUMMY_DEVICE_STATE, DUMMY_ELECTRIC_CURRENT, DUMMY_IP_ADDRESS,
+    DUMMY_MAC_ADDRESS, DUMMY_PHONE_ID, DUMMY_POWER_CONSUMPTION,
+    DUMMY_REMAINING_TIME, FULL_CONFIGURATION, MANDATORY_CONFIGURATION)
+
+
+async def test_failed_config(hass: HomeAssistant) -> None:
+    """Test failed configuration."""
+    assert await async_setup_component(
+        hass, DOMAIN, MANDATORY_CONFIGURATION) is False
+
+
+# pylint: disable=redefined-outer-name
+async def test_minimal_config(hass: HomeAssistant,
+                              mock_bridge_fixture: Generator[None, Any, None]
+                              ) -> None:
+    """Test setup with configuration minimal entries."""
+    with assert_setup_component(4, DOMAIN):
```

Comment:
Where does 4 (and 5 below) come from?

Reply:
When I test the minimum mandatory configuration keys:
- phone_id
- device_id
- device_password

The number of keys for the DOMAIN configuration will be 4 (not 3) because the name
key has a default value (the icon key does not). 

So the expected result will be 4.

When I test the full allowed configuration keys:
- phone_id
- device_id
- device_password
- name
- icon

The expected result will of course be 5.

Reply:
Fixed in this [commit](https://github.com/home-assistant/home-assistant/pull/22325/commits/1c8199d412e3e63a0df10e604288f565d8a6ca6f).